migrations/20220822175311_user-settings.sql

```diff
@@ -0,0 +1,9 @@
+CREATE TABLE user_settings (
+    user_id bigint NOT NULL PRIMARY KEY UNIQUE,
+    public_github boolean NOT NULL DEFAULT TRUE,
+    theme varchar(32) NOT NULL DEFAULT 'system',
+    locale varchar(32) NOT NULL DEFAULT 'en-US'
```

Comment:
Hey, in my Knossos i18n branch I have made automatic locale detection through `Accept-Language` header and `navigator.languages` browser API. It's also provided as one of the options in languages menu. In Nuxt plugin API it is not represented in any way, but you can change language to `'auto'` which internally switches to browser language / default locale (if detection failed) and turns on automatic mode, it then removes all stored language preferences since automatic detection is the default [^1].

![Screenshot highlighting ‘Sync with browser’ option for languages](https://user-images.githubusercontent.com/10401817/193461633-9b3c2b96-ffea-43f2-adda-f25c3ad48467.png)

That being said, I'd assume it'd make sense for this default here to also be `'auto'` (special reserved value) or nullable instead, where null would represent absence of locale preference, thus making applications figure out which locale to use.
 
[^1]: [`changeLanguage` implementation](https://github.com/Brawaru/knossos/blob/feat/i18n-rewrite/modules/i18n/templates/plugin.main.js#L312-L336)